### ISSUE TITLE
battle.net 

### ISSUE THREAD

**Issue by briceburg** [wine-compholio/wine-staging#299]
I am using a fresh install of wine-staging from the pipelight repository on an multilib archlinux system. Using the amd catalyst drivers (fglrx).

The battle.net application from Blizzard crashes.  

```
/opt/wine-staging/bin/wine Battle.net\ Launcher.exe
```

The battle.net login dialog appears, I enter my credentials, then the application crashes.  
Unable to install hearthstone on the system.

**NOTE**:
Upon first launching wine it asked me to download the mono .net framework, which I did from the wine dialog (did not install from archlinux packages).

Any help is much appreciated.

System
----------
```
 Wine build: wine-1.7.34 (Staging)
    Platform: i386 (WOW64)
    Host system: Linux
    Host version: 3.17.6-1-ARCH
```

stdout
--------

```
fixme:winhttp:get_system_proxy_autoconfig_url no support on this platform
fixme:toolhelp:CreateToolhelp32Snapshot Unimplemented: heap list snapshot
fixme:toolhelp:Heap32ListFirst : stub
fixme:wbemprox:enum_class_object_Next timeout not supported
err:ole:CoGetClassObject class {77f10cf0-3db5-4966-b520-b7c54fd35ed6} not registered
err:ole:CoGetClassObject no class object {77f10cf0-3db5-4966-b520-b7c54fd35ed6} could be created for context 0x1
fixme:wbemprox:enum_class_object_Next timeout not supported
fixme:ntdll:find_reg_tz_info Can't find matching timezone information in the registry for bias 282, std (d/m/y): 1/11/2015, dlt (d/m/y): 8/03/2015
fixme:winhttp:get_system_proxy_autoconfig_url no support on this platform
fixme:wtsapi:WTSRegisterSessionNotification Stub 0x100ae 0x00000000
fixme:d3d9:D3DPERF_GetStatus (void) : stub
fixme:winhttp:get_system_proxy_autoconfig_url no support on this platform
fixme:winhttp:get_system_proxy_autoconfig_url no support on this platform
fixme:winhttp:get_system_proxy_autoconfig_url no support on this platform
wine: Unhandled page fault on read access to 0x00000000 at address 0x7ce86730 (thread 0030), starting debugger...
```

Backtrace
-------------

```
Unhandled exception: assertion failed in 32-bit code (0xf7744bf0).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:f7744bf0 ESP:00338fe0 EBP:7c0c6a7c EFLAGS:00000296(   - --  I S -A-P- )
 EAX:00000000 EBX:000021da ECX:000021da EDX:00000006
 ESI:f7350000 EDI:f7510000
Stack dump:
0x00338fe0:  7c0c6a7c 00000006 000021da f7387f67
0x00338ff0:  f7510000 00339090 f73895a9 00000006
0x00339000:  00339010 00000000 f73c63a0 00000000
0x00339010:  00000020 00000000 00000000 00000000
0x00339020:  00000000 00000000 00000000 00000000
0x00339030:  00000000 00000000 00000000 00000000
000c: sel=0067 base=00000000 limit=00000000 16-bit rw-
Backtrace:
=>0 0xf7744bf0 __kernel_vsyscall+0x10() in [vdso].so (0x7c0c6a7c)
  1 0xf7387f67 gsignal+0x46() in libc.so.6 (0x7c0c6a7c)
  2 0xf73895a9 abort+0x148() in libc.so.6 (0x7c0c6a7c)
  3 0xf7381027 __assert_fail_base+0x116() in libc.so.6 (0x7c0c6a7c)
  4 0xf73810ab __assert_fail+0x3a() in libc.so.6 (0x7c0c6a7c)
  5 0x7cedc3d1 _XReply+0x420() in libx11.so.6 (0x7c0c6a7c)
  6 0x7ced18e6 XQueryExtension+0x95() in libx11.so.6 (0x7c0c6a7c)
  7 0x7cec5abe XInitExtension+0x3d() in libx11.so.6 (0x7c0c6a7c)
  8 0x7cfe6d6c XextAddDisplay+0x4b() in libxext.so.6 (0x7c0df364)
  9 0x7c0b9515 in libgl.so.1 (+0x5c514) (0x7ca9b228)
  10 0x7c1359b0 (0xf7510c08)
  11 0xf7510c00 main_arena+0x3bf() in libc.so.6 (0xf7510c00)
  12 0x7caa1498 (0x7caa1498)
0xf7744bf0 __kernel_vsyscall+0x10 in [vdso].so: popl	%ebp
Modules:
Module	Address			Debug info	Name (173 modules)
PE	  350000-  3db000	Deferred        qt5multimedia
PE	  400000-  db1000	Deferred        battle.net
PE	  dc0000- 10eb000	Deferred        qt5gui
PE	 10f0000- 11aa000	Deferred        libglesv2
PE	 11b0000- 1421000	Deferred        qt5quick
PE	 1430000- 1463000	Deferred        qt5winextras
PE	 1f70000- 2054000	Deferred        qwindows
PE	 2060000- 2085000	Deferred        libegl
PE	 2dc0000- 2dc9000	Deferred        qgif
PE	 2ee0000- 2ee9000	Deferred        qico
PE	 2ef0000- 2f26000	Deferred        qjpeg
PE	 2f30000- 2f6b000	Deferred        qmng
PE	 2f70000- 2f78000	Deferred        qsvg
PE	 34a0000- 34d7000	Deferred        qt5svg
PE	 34e0000- 3531000	Deferred        qtiff
PE	 3550000- 3757000	Deferred        d3dcompiler_43
PE	 6440000- 6446000	Deferred        qtquick2plugin
PE	 6bc0000- 6bd1000	Deferred        qquicklayoutsplugin
PE	 9cd0000- 9cd6000	Deferred        modelsplugin
PE	10000000-11921000	Deferred        libcef
PE	3c910000-3d14c000	Deferred        battle.net
PE	4ad00000-4b681000	Deferred        icudt
PE	61000000-6102a000	Deferred        qt5xml
PE	64000000-640b6000	Deferred        qt5network
PE	65000000-65448000	Deferred        qt5widgets
PE	66000000-66287000	Deferred        qt5qml
PE	67000000-67473000	Deferred        qt5core
ELF	7827a000-7a800000	Deferred        fglrx_dri.so
ELF	7a800000-7a91d000	Deferred        opengl32<elf>
  \-PE	7a820000-7a91d000	\               opengl32
ELF	7b800000-7ba5e000	Deferred        kernel32<elf>
  \-PE	7b810000-7ba5e000	\               kernel32
ELF	7bc00000-7bceb000	Deferred        ntdll<elf>
  \-PE	7bc10000-7bceb000	\               ntdll
ELF	7bf00000-7bf04000	Deferred        <wine-loader>
ELF	7bfa9000-7c05d000	Deferred        libatiadlxx.so
ELF	7c05d000-7c100000	Dwarf           libgl.so.1
ELF	7c433000-7c450000	Deferred        libatiuki.so.1
ELF	7c450000-7c456000	Deferred        libattr.so.1
ELF	7c47f000-7c499000	Deferred        cryptnet<elf>
ELF	7c499000-7c500000	Deferred        dbghelp<elf>
ELF	7ca01000-7ca18000	Deferred        libnss_myhostname.so.2
ELF	7cafd000-7cb06000	Deferred        librt.so.1
ELF	7cb06000-7cb49000	Deferred        rsaenh<elf>
  \-PE	7cb10000-7cb49000	\               rsaenh
ELF	7cb49000-7cc00000	Deferred        msvcrt<elf>
  \-PE	7cb60000-7cc00000	\               msvcrt
ELF	7cd05000-7cd1f000	Deferred        imagehlp<elf>
ELF	7cd79000-7cd80000	Deferred        libnss_dns.so.2
ELF	7cd80000-7cd9f000	Deferred        wintab32<elf>
  \-PE	7cd90000-7cd9f000	\               wintab32
ELF	7cd9f000-7cdb5000	Deferred        dwmapi<elf>
  \-PE	7cda0000-7cdb5000	\               dwmapi
ELF	7cdd7000-7cdf2000	Deferred        libgcc_s.so.1
ELF	7cdf2000-7cdf8000	Deferred        libtxc_dxtn.so
ELF	7cdf8000-7ce2e000	Deferred        uxtheme<elf>
  \-PE	7ce00000-7ce2e000	\               uxtheme
ELF	7ce2e000-7ce34000	Deferred        libxfixes.so.3
ELF	7ce34000-7ce3f000	Deferred        libxcursor.so.1
ELF	7ce3f000-7ce52000	Deferred        libxi.so.6
ELF	7ce52000-7ce56000	Deferred        libxcomposite.so.1
ELF	7ce56000-7ce61000	Deferred        libxrandr.so.2
ELF	7ce61000-7ce6c000	Deferred        libxrender.so.1
ELF	7ce6c000-7ce72000	Deferred        libxxf86vm.so.1
ELF	7ce72000-7ce79000	Deferred        libxdmcp.so.6
ELF	7ce79000-7ce7d000	Deferred        libxau.so.6
ELF	7ce7d000-7cea3000	Deferred        libxcb.so.1
ELF	7cea3000-7cfda000	Dwarf           libx11.so.6
ELF	7cfda000-7cfef000	Dwarf           libxext.so.6
ELF	7cfef000-7d081000	Deferred        winex11<elf>
  \-PE	7d000000-7d081000	\               winex11
ELF	7d081000-7d0f4000	Deferred        libgmp.so.10
ELF	7d0f4000-7d123000	Deferred        libhogweed.so.2
ELF	7d123000-7d156000	Deferred        libnettle.so.4
ELF	7d156000-7d16a000	Deferred        libtasn1.so.6
ELF	7d16a000-7d1ce000	Deferred        libp11-kit.so.0
ELF	7d1ce000-7d307000	Deferred        libgnutls.so.28
ELF	7d307000-7d30c000	Deferred        libcap.so.2
ELF	7d37f000-7d3a8000	Deferred        libexpat.so.1
ELF	7d3a8000-7d3e4000	Deferred        libfontconfig.so.1
ELF	7d3e4000-7d457000	Deferred        libpcre.so.1
ELF	7d457000-7d57e000	Deferred        libglib-2.0.so.0
ELF	7d57e000-7d5db000	Deferred        libharfbuzz.so.0
ELF	7d5db000-7d61a000	Deferred        libpng16.so.16
ELF	7d61a000-7d62b000	Deferred        libbz2.so.1.0
ELF	7d62b000-7d6f7000	Deferred        libfreetype.so.6
ELF	7d6f9000-7d702000	Deferred        libffi.so.6
ELF	7d720000-7d793000	Deferred        ddraw<elf>
  \-PE	7d730000-7d793000	\               ddraw
ELF	7d793000-7d7ab000	Deferred        wtsapi32<elf>
  \-PE	7d7a0000-7d7ab000	\               wtsapi32
ELF	7d7ab000-7d7cb000	Deferred        dnsapi<elf>
  \-PE	7d7b0000-7d7cb000	\               dnsapi
ELF	7d7cb000-7d90a000	Deferred        wined3d<elf>
  \-PE	7d7e0000-7d90a000	\               wined3d
ELF	7d90a000-7d948000	Deferred        d3d9<elf>
  \-PE	7d910000-7d948000	\               d3d9
ELF	7d948000-7d96c000	Deferred        imm32<elf>
  \-PE	7d950000-7d96c000	\               imm32
ELF	7d96c000-7da6a000	Deferred        comctl32<elf>
  \-PE	7d970000-7da6a000	\               comctl32
ELF	7da6a000-7db51000	Deferred        comdlg32<elf>
  \-PE	7da70000-7db51000	\               comdlg32
ELF	7db51000-7db90000	Deferred        winspool<elf>
  \-PE	7db60000-7db90000	\               winspool
ELF	7db90000-7dbbb000	Deferred        msacm32<elf>
  \-PE	7dba0000-7dbbb000	\               msacm32
ELF	7dbbb000-7dc72000	Deferred        winmm<elf>
  \-PE	7dbc0000-7dc72000	\               winmm
ELF	7dc72000-7dc86000	Deferred        psapi<elf>
  \-PE	7dc80000-7dc86000	\               psapi
ELF	7dc86000-7dcca000	Deferred        usp10<elf>
  \-PE	7dc90000-7dcca000	\               usp10
ELF	7dcca000-7dcf9000	Deferred        netapi32<elf>
  \-PE	7dcd0000-7dcf9000	\               netapi32
ELF	7dcf9000-7dd2a000	Deferred        secur32<elf>
  \-PE	7dd00000-7dd2a000	\               secur32
ELF	7dd2a000-7dd51000	Deferred        mpr<elf>
  \-PE	7dd30000-7dd51000	\               mpr
ELF	7dd51000-7ddcc000	Deferred        wininet<elf>
  \-PE	7dd60000-7ddcc000	\               wininet
ELF	7ddcc000-7de6a000	Deferred        urlmon<elf>
  \-PE	7dde0000-7de6a000	\               urlmon
ELF	7de6a000-7de91000	Deferred        iphlpapi<elf>
  \-PE	7de70000-7de91000	\               iphlpapi
ELF	7de91000-7dfcd000	Deferred        oleaut32<elf>
  \-PE	7deb0000-7dfcd000	\               oleaut32
ELF	7dfcd000-7e00b000	Deferred        winhttp<elf>
  \-PE	7dfd0000-7e00b000	\               winhttp
ELF	7e00b000-7e022000	Deferred        libz.so.1
ELF	7e022000-7e219000	Deferred        libcrypto.so.1.0.0
ELF	7e219000-7e289000	Deferred        libssl.so.1.0.0
ELF	7e289000-7e2a2000	Deferred        libresolv.so.2
ELF	7e2a2000-7e2b2000	Deferred        liblber-2.4.so.2
ELF	7e2b2000-7e30b000	Deferred        libldap_r-2.4.so.2
ELF	7e317000-7e334000	Deferred        jsproxy<elf>
  \-PE	7e320000-7e334000	\               jsproxy
ELF	7e334000-7e38f000	Deferred        wldap32<elf>
  \-PE	7e340000-7e38f000	\               wldap32
ELF	7e38f000-7e45e000	Deferred        crypt32<elf>
  \-PE	7e3a0000-7e45e000	\               crypt32
ELF	7e45e000-7e494000	Deferred        wintrust<elf>
  \-PE	7e460000-7e494000	\               wintrust
ELF	7e494000-7e55e000	Deferred        msvcr100<elf>
  \-PE	7e4b0000-7e55e000	\               msvcr100
ELF	7e55e000-7e66d000	Deferred        msvcp100<elf>
  \-PE	7e590000-7e66d000	\               msvcp100
ELF	7e66d000-7e6a6000	Deferred        ws2_32<elf>
  \-PE	7e670000-7e6a6000	\               ws2_32
ELF	7e6a6000-7e727000	Deferred        rpcrt4<elf>
  \-PE	7e6b0000-7e727000	\               rpcrt4
ELF	7e727000-7e85e000	Deferred        ole32<elf>
  \-PE	7e740000-7e85e000	\               ole32
ELF	7e85e000-7e8d5000	Deferred        shlwapi<elf>
  \-PE	7e870000-7e8d5000	\               shlwapi
ELF	7e8d5000-7eb05000	Deferred        shell32<elf>
  \-PE	7e8e0000-7eb05000	\               shell32
ELF	7eb05000-7eb75000	Deferred        advapi32<elf>
  \-PE	7eb10000-7eb75000	\               advapi32
ELF	7eb75000-7ec94000	Deferred        gdi32<elf>
  \-PE	7eb80000-7ec94000	\               gdi32
ELF	7ec94000-7ee01000	Deferred        user32<elf>
  \-PE	7ecb0000-7ee01000	\               user32
ELF	7ef8a000-7efd7000	Deferred        libm.so.6
ELF	7efe6000-7f000000	Deferred        version<elf>
  \-PE	7eff0000-7f000000	\               version
ELF	f7357000-f735c000	Deferred        libdl.so.2
ELF	f735c000-f7514000	Dwarf           libc.so.6
ELF	f7514000-f7531000	Deferred        libpthread.so.0
ELF	f7532000-f7540000	Deferred        libnss_files.so.2
ELF	f755a000-f771e000	Dwarf           libwine.so.1
ELF	f771f000-f7742000	Deferred        ld-linux.so.2
ELF	f7744000-f7745000	Dwarf           [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
0000000e services.exe
	00000022    0
	0000001c    0
	00000016    0
	00000014    0
	00000010    0
	0000000f    0
00000012 winedevice.exe
	0000001b    0
	00000018    0
	00000017    0
	00000013    0
00000019 plugplay.exe
	0000001f    0
	0000001e    0
	0000001a    0
00000020 explorer.exe
	00000021    0
0000002e Agent.exe
	0000005b    0
	00000046    0
	00000041    0
	00000040    0
	0000003f    0
	0000003e    0
	0000003d    0
	0000003c    0
	0000003b    0
	00000039    0
	00000035    0
	0000002f    0
00000031 wineconsole.exe
	00000032    0
00000028 (D) C:\Program Files (x86)\Battle.net\Battle.net.5383\Battle.net.exe
	00000030    0 <==
System information:
    Wine build: wine-1.7.34 (Staging)
    Platform: i386 (WOW64)
    Host system: Linux
    Host version: 3.17.6-1-ARCH
```

kernel modules
--------------------

```
Module                  Size  Used by
vmnet                  50206  13 
fuse                   87410  3 
vmw_vsock_vmci_transport    29950  0 
vsock                  30817  1 vmw_vsock_vmci_transport
vmw_vmci               55463  1 vmw_vsock_vmci_transport
vmmon                  81251  0 
iptable_filter         12536  0 
ip_tables              26011  1 iptable_filter
x_tables               27104  2 ip_tables,iptable_filter
veth                   13140  0 
nls_iso8859_1          12461  0 
nls_cp437              16553  0 
vfat                   21231  0 
fat                    61984  1 vfat
isofs                  38966  0 
joydev                 17063  0 
hid_logitech_dj        17192  0 
uas                    21995  0 
usb_storage            60311  1 uas
bridge                114196  0 
stp                    12437  1 bridge
llc                    12745  2 stp,bridge
cfg80211              445286  0 
rfkill                 18867  1 cfg80211
mousedev               17272  0 
hid_generic            12393  0 
snd_usb_audio         143556  2 
usbhid                 48595  0 
snd_usbmidi_lib        27571  1 snd_usb_audio
hid                   102220  3 hid_generic,usbhid,hid_logitech_dj
snd_rawmidi            26806  1 snd_usbmidi_lib
snd_seq_device         13132  1 snd_rawmidi
ext4                  501833  1 
crc16                  12343  1 ext4
mbcache                17171  1 ext4
jbd2                   86417  1 ext4
coretemp               12820  0 
hwmon                  12930  1 coretemp
intel_rapl             17356  0 
x86_pkg_temp_thermal    12951  0 
intel_powerclamp       17122  0 
kvm_intel             143245  0 
kvm                   421519  1 kvm_intel
crct10dif_pclmul       13394  0 
crc32_pclmul           12915  0 
crc32c_intel           21809  0 
snd_hda_codec_hdmi     49213  1 
ghash_clmulni_intel    12978  0 
snd_hda_intel          26387  1 
mac_hid                12633  0 
aesni_intel           163855  0 
snd_hda_controller     26938  1 snd_hda_intel
fglrx               10873511  103 
aes_x86_64             16719  1 aesni_intel
lrw                    12757  1 aesni_intel
snd_hda_codec         108536  3 snd_hda_codec_hdmi,snd_hda_intel,snd_hda_controller
snd_hwdep              17244  2 snd_usb_audio,snd_hda_codec
snd_pcm                88487  5 snd_usb_audio,snd_hda_codec_hdmi,snd_hda_codec,snd_hda_intel,snd_hda_controller
snd_timer              26614  1 snd_pcm
snd                    73436  16 snd_usb_audio,snd_hwdep,snd_timer,snd_hda_codec_hdmi,snd_pcm,snd_rawmidi,snd_usbmidi_lib,snd_hda_codec,snd_hda_intel,snd_seq_device
iTCO_wdt               12831  0 
soundcore              13031  2 snd,snd_hda_codec
gf128mul               12970  1 lrw
iTCO_vendor_support    12649  1 iTCO_wdt
glue_helper            12649  1 aesni_intel
ablk_helper            12572  1 aesni_intel
cryptd                 18553  3 ghash_clmulni_intel,aesni_intel,ablk_helper
evdev                  21544  14 
ppdev                  16782  0 
psmouse               107214  0 
r8169                  68207  0 
amd_iommu_v2           17457  1 fglrx
serio_raw              12849  0 
pcspkr                 12595  0 
mii                    12675  1 r8169
battery                17452  0 
thermal                17559  0 
processor              27777  0 
parport_pc             26351  0 
parport                35749  2 ppdev,parport_pc
video                  18043  0 
button                 12953  1 fglrx
tpm_infineon           16882  0 
tpm_tis                17182  0 
tpm                    31467  2 tpm_tis,tpm_infineon
fan                    12681  0 
i2c_i801               16965  0 
mei_me                 17900  0 
i2c_core               50152  1 i2c_i801
mei                    75079  1 mei_me
shpchp                 35210  0 
lpc_ich                20768  0 
ie31200_edac           12567  0 
edac_core              47125  1 ie31200_edac
sch_fq_codel           17343  6 
f2fs                  199972  1 
sd_mod                 44398  3 
sr_mod                 21903  0 
crc_t10dif             12431  1 sd_mod
crct10dif_common       12356  2 crct10dif_pclmul,crc_t10dif
cdrom                  47479  1 sr_mod
ata_generic            12490  0 
pata_acpi              12771  0 
atkbd                  22254  0 
libps2                 12739  2 atkbd,psmouse
ata_piix               33592  2 
libata                181518  3 pata_acpi,ata_generic,ata_piix
ehci_pci               12512  0 
xhci_hcd              152911  0 
ehci_hcd               69939  1 ehci_pci
scsi_mod              147543  5 uas,usb_storage,libata,sd_mod,sr_mod
usbcore               199381  8 uas,snd_usb_audio,usb_storage,snd_usbmidi_lib,ehci_hcd,ehci_pci,usbhid,xhci_hcd
usb_common             12440  1 usbcore
i8042                  18002  1 libps2
serio                  18282  6 serio_raw,atkbd,i8042,psmouse
```

**Comment by slackner** [comment]
I assume you have missed https://github.com/wine-compholio/wine-staging/issues/292, we do no longer accept bug reports in this old bug tracker. All remaining bugs in this one will be closed or moved during the next weeks.

Some things I noticed in the logs above:

* Generally 64-bit wine support is still a bit experimental. You should create a new wineprefix with `WINEARCH=win32`, and then try again.

* The logs you provided above (terminal output and backtrace) do not match. Most likely you ran the application several times, and then combined the output from different runs. This makes the logs basically useless.

* Based on the log above the crash is somewhere in your graphic card driver. You could theoretically try with the mesa driver instead, maybe this solves the issue.

* Does the Battle.net Launcher work with plain wine? When the problem also occurs with plain wine (and can be reproduced without the hearthstone fix) it is more useful to open a bug report on the upstream wine bugtracker (http://bugs.winehq.org, make sure to only attach logs created with plain Wine there).

Please open a new bug report at http://bugs.wine-staging.com if you need additional help and/or think this issue is related to our patches.

**Comment by briceburg** [comment]
@slackner thank you for taking a look at this, and my apologies for reporting here. Re: stdout != backtrace, you're right -- I did run it again to capture the stdout thinking it may be helpful.

I am glad to report that the issue related to missing lib32 versions of mesa/opengl libraries. After installing the `lib32-catalyst-libgl` package from archlinux AUR, all seems well again.

The crash happened in wine 1.7.35, so I tried wine-staging. After suffering the same issue I installed playonlinux and it warned of missing 32bit opengl libraries, which led me to the fix.  

Super excited to try this out. Thanks!
